proof_test.go

```diff
@@ -0,0 +1,160 @@
+package merkletree
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+func computeLeafHash(ap *AuthenticationPath) (leafHash []byte) {
+	leaf := ap.Leaf()
+	if leaf.Commitment() != nil {
```

Comment:
This comparison makes me a little nervous.  Maybe you could implement `isEmpty` on `ProofNode` as well.